### ISSUE TITLE
[Datepicker]: wcag issues

### ISSUE THREAD

**Issue by kristiina-richards** [TEDI-Design-System/react#340]
### Description of the problem

1. When entering an invalid date, then it should show user an error stating that. Currently the date is kept visible with no error.
**Requirement:** [link](https://www.w3.org/WAI/WCAG22/Understanding/error-suggestion.html)
2. Date should have focus style
**Requirement:** [link](https://www.w3.org/WAI/WCAG22/Understanding/focus-visible.html)
3. JAWS doesn't read dates in calendar as clickable elements
**Requirement:** [link](https://www.w3.org/WAI/WCAG22/Understanding/name-role-value.html)
4. JAWS doesnt move to calendar < and > buttons. 
**Requirement:** [link](https://www.w3.org/WAI/WCAG22/Understanding/name-role-value.html)
5. JAWS loeb kalendris kuupäevad ette nt "Teisipäev, 1, column 2". Ekraanilugeja ei ütle, mis kuupäevaga tegemist on. Ekraanilugeja peab edasi andma sama informatsiooni, mis on visuaalselt näha. Ekraanilugeja peaks ütlema nt "Teisipäev, 2 Aprill 2025".
**Nõue:** https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships.html
